### ISSUE TITLE
Equals generation for same class

### ISSUE THREAD

**Issue by mayras** [mapstruct/mapstruct#1800]
it will be nice to have generation like method "equals" with ability to choose comparing fields or/and exclude such fields to replace EqualsBuilder.reflectionEquals Apache Commons Lang and increase performance.

**Comment by chris922** [comment]
Hi mayras,

I am sorry, I can't follow you. Maybe you could outline a short example? Do you mean that MapStruct should generate `equals` methods for your POJOs?

**Comment by filiphr** [comment]
There is [@EqualsAndHashCode](https://projectlombok.org/features/EqualsAndHashCode) from Lombok that would generate exactly what you are looking for (although on the byte code level).

I am going to close this issue as I don't think that this is within the scope of this project.

**Comment by mayras** [comment]
Everybody knows that method “equals“ b/w objects of same class can be different and it depends on tasks - in some case you use 2 fields, in another case you use 5 fields and so on, for better development - better to extract method "equals" out and place different implmenetation, it's why I wrote this ticket. For example:

@Mappings({
    @Mapping(source = 'color', ignore = true)
} )
boolean catEquals(Cat c1, Cat c2)

I can create different variations of "equals" and use what I like in necessary cases. I think it's very very useful.

**Comment by filiphr** [comment]
You are right it could be useful and it is an interesting use case. 

However, in my opinion this would belong in another project and not in MapStruct. The annotations in MapStruct are designed for mapping and for this use case you want to have something else. 

I'll leave the issue closed since this is out of scope for this project.

**Comment by mayras** [comment]
sad :(
